lib/shifts.rb

```diff
@@ -2,38 +2,36 @@
 class Shifts
   attr_reader :key, :date
 
-  def initialize(key = rand(99999), date = Date.today.strftime("%d%m%y"))
+  def initialize(key, date)
     @key = key
     @date = date
   end
 
-  def shifts(key = @key, date = @date)
-    key_array = shift_keys(zero_pad(key))
-    offsets = produce_offsets(date)
+  def shifts
     shifts = []
-    offsets.each_with_index do |element, index|
-      shifts << element.to_i + key_array[index].to_i
+    produce_offsets.each_with_index do |element, index|
+      shifts << element.to_i + shift_keys[index].to_i
     end
     shifts
   end
 
-  def zero_pad(key = @key)
-    pad_num = key.to_s
-    pad_num.insert(0, "0") until pad_num.size == 5
-    pad_num
-  end
-
-  def shift_keys(key = @key)
+  def shift_keys
```

Comment:
refactored most methods within `Shifts` to function off of instance variables rather than arguments.